package.json

```diff
@@ -20,8 +20,14 @@
     "types/index.d.ts"
   ],
   "exports": {
-    ".": "./index.js",
-    "./promise": "./promise.js"
+    ".": {
+      "default": "./index.js",
+      "types": "./types/index.d.ts"
+    },
+    "./promise": {
+      "default": "./promise.js",
+      "types": "./types/index.d.ts"
```

Comment:
For this I think we actually need a `types/promise.d.ts`. It's would be almost the same as `index.d.ts`, but with some of the return types as promises. @fehnomenal let me know if you will be able to work on this. If not, I can also do it.

Reply:
Yeah, you're right. I will do it later